ed25519/src/lib.rs

```diff
@@ -292,9 +292,7 @@ impl Signature {
     pub const BYTE_SIZE: usize = 64;
 
     /// Parse an Ed25519 signature from a byte slice.
-    pub fn from_bytes(bytes: &[u8]) -> signature::Result<Self> {
-        let result = bytes.try_into().map(Self).map_err(|_| Error::new())?;
-
+    pub fn from_bytes(bytes: [u8; Self::BYTE_SIZE]) -> signature::Result<Self> {
```

Comment:
Not taking a reference will use up more stack space making a copy where it could just be a pointer, unless LLVM inlines it.

Elsewhere we generally take a reference for this reason.